geode-core/src/test/java/org/apache/geode/internal/cache/tx/DistributedTXRegionStubTest.java

```diff
@@ -0,0 +1,371 @@
+import org.apache.geode.cache.TransactionException;
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.distributed.internal.membership.InternalDistributedMember;
+import org.apache.geode.internal.cache.DistributedPutAllOperation;
+import org.apache.geode.internal.cache.DistributedRegion;
+import org.apache.geode.internal.cache.DistributedRemoveAllOperation;
+import org.apache.geode.internal.cache.EntryEventImpl;
+import org.apache.geode.internal.cache.EventID;
+import org.apache.geode.internal.cache.KeyInfo;
+import org.apache.geode.internal.cache.LocalRegion;
+import org.apache.geode.internal.cache.RemoteOperationException;
+import org.apache.geode.internal.cache.TXStateStub;
+import org.apache.geode.internal.cache.tier.sockets.ClientProxyMembershipID;
+import org.apache.geode.internal.cache.tier.sockets.VersionedObjectList;
+
+public class DistributedTXRegionStubTest {
+
+  private TXStateStub txStateStub;
+  private DistributedRegion distributedRegion;
+  private Object expectedObject;
+  private EntryEventImpl event;
+  private DistributedMember remoteTransactionHost;
+  private KeyInfo keyInfo;
+  private Set<InternalDistributedMember> failures;
+
+
+  @Before
+  public void setup() {
+    txStateStub = mock(TXStateStub.class);
+    distributedRegion = mock(DistributedRegion.class, RETURNS_DEEP_STUBS);
+    event = mock(EntryEventImpl.class);
+    expectedObject = new Object();
+    remoteTransactionHost = mock(InternalDistributedMember.class);
+    keyInfo = mock(KeyInfo.class);
+
+    when(txStateStub.getTarget()).thenReturn(remoteTransactionHost);
+    when(event.getEventId()).thenReturn(mock(EventID.class));
+    when(event.getRegion()).thenReturn(distributedRegion);
+    when(keyInfo.getKey()).thenReturn(expectedObject);
+  }
+
+  @Test
+  public void destroyExistingEntryTranslatesRegionDestroyedException() throws Exception {
+    DistributedTXRegionStub stub = spy(new DistributedTXRegionStub(txStateStub, distributedRegion));
+
+    when(event.getRegion()).thenThrow(RegionDestroyedException.class);
+
+    Throwable caughtException =
+        catchThrowable(() -> stub.destroyExistingEntry(event, true, expectedObject));
+
+    assertThat(caughtException).isInstanceOf(TransactionDataNotColocatedException.class);
+  }
+
+  @Test
+  public void destroyExistingEntryTranslatesRemoteOperationException() throws Exception {
+    DistributedTXRegionStub stub = spy(new DistributedTXRegionStub(txStateStub, distributedRegion));
+    when(event.getRegion()).thenReturn(distributedRegion);
+
+    doThrow(RemoteOperationException.class).when(stub).sendRemoteDestroyMessage(event,
+        expectedObject);
+
+    Throwable caughtException =
+        catchThrowable(() -> stub.destroyExistingEntry(event, true, expectedObject));
+
+    assertThat(caughtException).isInstanceOf(TransactionDataNodeHasDepartedException.class);
+  }
+
+  @Test
+  public void getEntryEntryNotFoundExceptionReturnNull() throws Exception {
```

Comment:
Exception is never thrown.

Also, will getEntryReturnsNullIfGotEntryNotFoundException be a better method name?